config/plugin.js

```diff
@@ -0,0 +1,55 @@
+const path = require('path')
+const { webpack } = require('next/dist/compiled/webpack/webpack')
+const { getHashiConfig } = require('./index')
+
+/**
+ * Reads in config files from config/[env].json and replaces references in the
+ * code with the literal values using webpack.DefinePlugin.
+ *
+ * Supports extending configs via the `extends` property.
+ *
+ * Example:
```

Comment:
Can you illustrate some deeply nested properties to show that we don't overwrite the entire object at that path?

Reply:
Added to the test case and referenced that in this docs comment, good call!